src/app/dialogs/edit-poll-item-dialog/edit-poll-item-dialog.component.ts

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright © Live-Poll 2020-2021. All rights reserved
+ */
+
+import {Component, EventEmitter, Input, OnInit, Output} from '@angular/core';
+import {OPTIONS_DATA, OptionType} from '../../shared/poll-item-options';
+import {Poll} from '../../model/poll';
+import {CommonToolsService} from '../../service/common-tools.service';
+import {PollItemService} from '../../service/poll-item.service';
+import {OpenTextItemCreate} from '../../model/poll-item-create/open-text-item-create';
+import {MultipleChoiceItemCreate} from '../../model/poll-item-create/multiple-choice-item-create';
+import {QuizItemCreate} from '../../model/poll-item-create/quiz-item-create';
+import {MultipleChoiceItemParticipant} from '../../model/poll-item-participant/multiple-choice-item-participant';
+import {OpenTextItemParticipant} from '../../model/poll-item-participant/open-text-item-participant';
+import {QuizItemParticipant} from '../../model/poll-item-participant/quiz-item-participant';
+
+// Constants
+const STEP_LABELS = [
+  'Update question / answers',
+  'Update options'
+];
+
+/**
+ * Wizard dialog, which lets the user edit an existing poll item.
+ * The dialog consists of two pages, whose appearance differ based on the selected item type.
+ */
+@Component({
+  selector: 'app-edit-poll-item-dialog',
+  templateUrl: './edit-poll-item-dialog.component.html',
+  styleUrls: ['./edit-poll-item-dialog.component.sass']
+})
+export class EditPollItemDialogComponent implements OnInit {
+
+  // Constant associations
+  stepLabels = STEP_LABELS;
+  optionTypes = OptionType;
+
+  // Event Emitters
+  @Input() isVisible: boolean;
+  @Input() poll: Poll;
+  @Input() pollItem: MultipleChoiceItemParticipant|OpenTextItemParticipant|QuizItemParticipant;
+  @Output() finish = new EventEmitter<boolean>(); // true = success; false = cancel
+
+  // Variables
+  step = 1;
+  loading = false;
+  errorMessage = '';
+  itemType = 0;
+  question = '';
+  answers = ['', ''];
```

Comment:
Why is `['', '']` the default value for `answers`? I see that you want at least two selection options for multiple choice and quiz items, but why is this the default value then?

Reply:
Exactly because of this

Reply:
ok